diff -uNr heroku-2.3.6.orig//lib/heroku/client.rb heroku-2.3.6/lib/heroku/client.rb
--- all/heroku-2.3.6.orig//lib/heroku/client.rb	2011-07-16 00:15:48.432332831 +0900
+++ all/heroku-2.3.6/lib/heroku/client.rb	2011-07-16 00:17:27.136822277 +0900
@@ -525,7 +525,7 @@
 
   def resource(uri, options={})
     RestClient.proxy = ENV['HTTP_PROXY'] || ENV['http_proxy']
-    resource = RestClient::Resource.new(realize_full_uri(uri), options.merge(:user => user, :password => password))
+    resource = RestClient::Resource.new(realize_full_uri(uri), options.merge(:user => user, :password => password, :ssl_ca_file => '/etc/ssl/certs/ca-certificates.crt'))
     resource
   end
 
