diff -uNr pam_ldap-185.orig/pam_ldap.c pam_ldap-185.new/pam_ldap.c
--- pam_ldap-185.orig/pam_ldap.c	2009-11-06 19:29:34.000000000 +0900
+++ pam_ldap-185.new/pam_ldap.c	2010-04-17 02:44:50.373911029 +0900
@@ -2488,17 +2488,9 @@
 _host_ok (pam_ldap_session_t * session)
 {
   char hostname[MAXHOSTNAMELEN];
-  struct hostent *h;
-#ifdef HAVE_GETHOSTBYNAME_R
-  struct hostent hbuf;
-#if GETHOSTBYNAME_R_ARGS == 3
-  struct hostent_data buf;
-#else
-  int herr;
-  char buf[1024];
-#endif /* GETHOSTBYNAME_R_ARGS == 3 */
-#endif /* HAVE_GETHOSTBYNAME_R */
-  char **q;
+  struct addrinfo hints;
+  struct addrinfo *result, *rp;
+  int s;
 
   /* simple host based access authorization */
   if (session->info->hosts_allow == NULL)
@@ -2512,54 +2504,30 @@
       return PAM_SYSTEM_ERR;
     }
 
-#if defined(HAVE_GETHOSTBYNAME_R)
-#if GETHOSTBYNAME_R_ARGS == 6
-  if (gethostbyname_r (hostname, &hbuf, buf, sizeof buf, &h, &herr) != 0)
-    {
-      return PAM_SYSTEM_ERR;
-    }
-#elif GETHOSTBYNAME_R_ARGS == 5
-  h = gethostbyname_r (hostname, &hbuf, buf, sizeof buf, &herr);
-  if (h == NULL)
-    {
-      return PAM_SYSTEM_ERR;
-    }
-#elif GETHOSTBYNAME_R_ARGS == 3
-  if (gethostbyname_r (hostname, &hbuf, &buf) != 0)
-    {
-      return PAM_SYSTEM_ERR;
-    }
-  h = &hbuf;
-#else
-#error Unknown gethostbyname_r() implementation
-#endif
-#else
-  h = gethostbyname (hostname);
-  if (h == NULL)
+  memset(&hints, 0, sizeof(struct addrinfo));
+  hints.ai_family    = AF_UNSPEC;	/* Allow IPv4 or IPv6 */
+  hints.ai_socktype  = 0;	/* Any Socket */
+  hints.ai_flags     = AI_CANONNAME;	/* Have Canonical Name */
+  hints.ai_protocol  = 0;	/* Any Protocol */
+  hints.ai_canonname = NULL;
+  hints.ai_addr      = NULL;
+  hints.ai_next      = NULL;
+
+  if (getaddrinfo(hostname, NULL, &hints, &result) != 0)
     {
-      return PAM_SYSTEM_ERR;
+	    return PAM_SYSTEM_ERR;
     }
-#endif
 
-  if (h == NULL || h->h_name == NULL) 
-    return PAM_SYSTEM_ERR; 
-
-  if (_has_deny_value (session->info->hosts_allow, h->h_name))
-    return PAM_PERM_DENIED;
-  else if (_has_value (session->info->hosts_allow, h->h_name))
-    return PAM_SUCCESS;
-
-  if (h->h_aliases != NULL)
+  for (rp = result; rp->ai_canonname != NULL; rp = rp->ai_next)
     {
-      for (q = h->h_aliases; *q != NULL; q++)
-	{
-	  if (_has_value (session->info->hosts_allow, *q))
-	    return PAM_SUCCESS;
-	  if (_has_deny_value (session->info->hosts_allow, *q))
-	    return PAM_PERM_DENIED;
-	}
+  	if (_has_deny_value (session->info->hosts_allow, rp->ai_canonname))
+  	  return PAM_PERM_DENIED;
+  	else if (_has_value (session->info->hosts_allow, rp->ai_canonname))
+  	  return PAM_SUCCESS;
     }
 
+  freeaddrinfo(result);
+
   /* allow wild-card entries */
   if (_has_value (session->info->hosts_allow, "*"))
     {
