--- configure.ac.orig	2010-07-07 21:05:43.749421847 +0900
+++ configure.ac	2010-07-07 21:07:49.568477805 +0900
@@ -148,11 +148,11 @@
 AC_SUBST(PIXMAN_LIBS)
 SPICE_REQUIRES+=" pixman-1 >= 0.17.7"
 
-PKG_CHECK_MODULES(CELT051, celt051 >= 0.5.1.1)
-AC_SUBST(CELT051_CFLAGS)
-AC_SUBST(CELT051_LIBS)
-AC_SUBST(CELT051_LIBDIR)
-SPICE_REQUIRES+=" celt051 >= 0.5.1.1"
+PKG_CHECK_MODULES(CELT, celt >= 0.5.1.1)
+AC_SUBST(CELT_CFLAGS)
+AC_SUBST(CELT_LIBS)
+AC_SUBST(CELT_LIBDIR)
+SPICE_REQUIRES+=" celt >= 0.5.1.1"
 
 PKG_CHECK_MODULES(ALSA, alsa)
 AC_SUBST(ALSA_CFLAGS)
@@ -164,26 +164,10 @@
 AC_SUBST(SSL_LIBS)
 SPICE_REQUIRES+=" openssl"
 
-# These are commented out because the gl libraries on RHEL 5 do not have pkgconfig files
-#
-# PKG_CHECK_MODULES(GL, gl glu)
-# AC_SUBST(GL_CFLAGS)
-# AC_SUBST(GL_LIBS)
-# SPICE_REQUIRES+=" gl glu"
-
-if test "x$have_opengl" = "xyes"; then
-   AC_CHECK_LIB(GL, glBlendFunc, GL_LIBS="$GL_LIBS -lGL", have_opengl=no)
-   AC_CHECK_LIB(GLU, gluSphere, GL_LIBS="$GL_LIBS -lGLU", have_opengl=no)
-   GL_CFLAGS="-DGL_GLEXT_PROTOTYPES -DUSE_OGL"
-
-   if test "x$have_opengl" = "xno"; then
-      AC_MSG_ERROR([GL libraries not available])
-   fi
-fi
-
+PKG_CHECK_MODULES(GL, gl glu)
 AC_SUBST(GL_CFLAGS)
 AC_SUBST(GL_LIBS)
-SPICE_NONPKGCONFIG_LIBS+=" $GL_LIBS"
+SPICE_REQUIRES+=" gl glu"
 
 PKG_CHECK_MODULES(XRANDR, xrandr)
 AC_SUBST(XRANDR_CFLAGS)
